### PR TITLE
Feature/wonder

### PR DESCRIPTION
**Pull request by anipmehta**

### CODE REVIEW

File: src/main/game/model/PlayerId.java [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 public class PlayerId {
+
     private static int currentPlayerId = 1;
 
     private int playerId;
     PlayerId() {
         this.playerId = currentPlayerId;
         currentPlayerId++;
     }
+    public PlayerId(int num){
```
Comment (gavinscheele) [inline src/main/game/model/PlayerId.java]: this shouldn't be here

File: src/main/game/model/wonder/Wonder.java [hunk 2]
```diff
@@ -1,26 +1,84 @@
...
+import game.model.Player;
+import game.model.PlayerId;
 
+import java.util.HashMap;
 import java.util.Vector;
 
 public class Wonder implements TurnObserver {
+    private Vector<WonderBrick> player1;
+    private Vector<WonderBrick> player2;
+    Vector<Vector<WonderBrick>> wonderbricks;
```
Comment (gavinscheele) [inline src/main/game/model/wonder/Wonder.java]: can you make this private

File: src/main/game/model/wonder/WonderManager.java [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-
-//    public int getBrickCost() {
-//
-//    }
 //
-//    public void turnDesertToPasture() {
-//
-//    }
+    public void turnDesertToPasture() {
+
```
Comment (gavinscheele) [inline src/main/game/model/wonder/WonderManager.java]: This still needs to be implemented

File: src/main/game/model/wonder/Wonder.java [hunk 4]
```diff
@@ -1,26 +1,84 @@
...
+            {
+                //max size of bricks in each row = 6
+                if (wonderBricks.size() < 6) {
+                    wonderBricks.add(new PlayerBrick(pid));
+                    flag = true;
+                    break;
+                }
+            }
+        }
+        if(!flag){
```
Comment (gavinscheele) [inline src/main/game/model/wonder/Wonder.java]: can you rename the flag variable to be more descriptive

File: src/main/game/model/wonder/Wonder.java [hunk 5]
```diff
@@ -1,26 +1,84 @@
...
+        //Checking whether the first 4 row is full or not
+        if(wonderbricks.size()<4) {
+                cost = playerIdVectorHashMap.get(playerId).size()+1;
+        }
+        else{
+                cost = playerIdVectorHashMap.get(playerId).size()+2;
+        }
+        return cost;
+    }
 }
```
Comment (gavinscheele) [inline src/main/game/model/wonder/Wonder.java]: We still need a method getWealthPoints(PlayerId) which will return the total wealth points that a player earned in the wonder. This needs to account for the number of bricks each player has in each row.

**Review by gavinscheele — CHANGES_REQUESTED** [review]

File: src/main/game/model/wonder/Wonder.java (hunk 2 above)
Comment (anipmehta) [reply src/main/game/model/wonder/Wonder.java]: @gdscheele Just to make sure you are talking about the wonderbicks right??

Comment (gavinscheele) [reply src/main/game/model/wonder/Wonder.java]: yeah

File: src/main/game/model/wonder/Wonder.java (hunk 4 above)
Comment (anipmehta) [reply src/main/game/model/wonder/Wonder.java]: Done

File: src/main/game/model/wonder/Wonder.java (hunk 5 above)
Comment (anipmehta) [reply src/main/game/model/wonder/Wonder.java]: Added

**Comment by anipmehta** [comment]
@gdscheele Only turnDesertToPasture is left else every review were implemented.

**Comment by anipmehta** [comment]
Added more test in Wonder stuff as suggested by @gdscheele .

**Comment by anipmehta** [comment]
@gdscheele I have implemented turnDesertToPasture functionality. Please review if that is a correct way or do I need to change anything.

**Comment by anipmehta** [comment]
@Juan-Jauregui Updated turnDesertToPasture functionality.

**Comment by anipmehta** [comment]
Wonder Functionality is almost complete. Pls review.
@tonny-12 All tests passed.

**Review by gavinscheele — DISMISSED** [review]

**Review by tonny-12 — APPROVED** [review]

**Pull request merged**